Gifski/TrimmingAVPlayerView.swift

```diff
@@ -0,0 +1,47 @@
+import AVKit
+
+final class TrimmingAVPlayerView: AVPlayerView {
+	private var timeRangeObserver: NSKeyValueObservation?
+
+	func observeTrimmedTimeRange(_ updateClosure: @escaping (ClosedRange<Double>) -> Void) {
+		timeRangeObserver = player?.currentItem?.observe(\.duration, options: [.new]) { item, _ in
+			let startTime = item.reversePlaybackEndTime.seconds
+			let endTime = item.forwardPlaybackEndTime.seconds
+			if !startTime.isNaN && !endTime.isNaN {
+				updateClosure(startTime...endTime)
+			}
+		}
+	}
+
+	func hideTrimButtons() {
+		// This method is a collection of hacks, so might be acting funky on different OS versions
+		let avTrimView = firstSubview(where: { $0.description.contains("AVTrimView") }, deep: true)
+
+		if let avTrimView = avTrimView {
```

Comment:
Use `guard` to reduce nesting. `let superview = avTrimView.superview` can be in the guard too.